### ISSUE TITLE
Support cipher suite configuration

### ISSUE THREAD

**Issue by vcsjones** [aspnet/AspNetCore#4776]
As I understand, Kestrel 2 will be considered "safe" for edge deployments with HTTPS. Currently, there is no way to easily configure many desirable parameters of HTTPS. Mainly, the cipher suites that are supported, the algorithm and key size of the key exchange, etc.

For Windows, the functionality is configured the same as it was with IIS - with the registry / group policy. For Linux, I'm not exactly sure how to change the configuration, or if it can be done at all. I suspect it will just use OpenSSL's default in the distribution. If that is not the case, then I would be keen in knowing how to do it.

The Windows configuration by registry for SChannel leaves a lot to be desired. Changing SChannel's default configuration is not just for IIS or Kestrel - it affects anything that uses SChannel (remote desktop for example).

For edge deployments, it is naturally desirable to want to configure the cipher suites. There is a lot of pressure and mandate to start turning off old versions of TLS and strengthening cipher suite configuration, ensuring poor symmetric cipher are disabled, like 3DES, etc. Setting a preferred order is also important. Most web servers will prioritize AES128 over AES256 for performance, while some customers may wish to trade off the performance for security.

Even for internal traffic use, this is still very useful. Internal TLS is important, and because the traffic is internal to my infrastructure, I can mandate it to be whatever I want, such as TLS 1.2 only and ECDHE-AES-GCM-128 and not have to worry about client compatibility.

**Comment by TachyonRSA** [comment]
Is there any feedback on this?

**Comment by avenema** [comment]
Is this unblocked now that https://github.com/dotnet/corefx/issues/24588 is resolved?

**Comment by shirhatti** [comment]
@avenema Yes. We've filed https://github.com/aspnet/AspNetCore/issues/9349 to track this work.

Closing this discussion issue in favor of the new issue
